src/coverageLib/FileCoverage.ts

```diff
@@ -0,0 +1,368 @@
+import { Stmt, Expr } from "../parser";
+import { Location, isToken, Lexeme } from "../lexer";
+import { BrsInvalid, BrsType } from "../brsTypes";
+import { isStatement } from "../parser/Statement";
+
+interface StatementCoverage {
+    /** Number of times the interpreter has executed/evaluated this statement. */
+    hits: number;
+    /** Combine expressions and statements because we need to log some expressions to get the coverage report. */
+    statement: Expr.Expression | Stmt.Statement;
+}
+
+export interface CoverageSummary {
+    path: string;
+    statements: { [key: string]: { hits: number; location: Location } };
+    functions: {
+        [key: string]: { hits: number; location: Location; name: string };
+    };
+    branches: {
+        [key: string]: {
+            hits: number[];
+            locations: Location[];
+            /** Branch type, i.e. "if", "or", "and", etc. */
+            type: string;
+        };
+    };
+}
+
+export class FileCoverage {
```

Comment:
Yuuuup definitely should do that

Reply:
This didn't _quite_ work out as cleanly as I hoped, because `AstNode` and `Statement`/`Expression` have no knowledge of each other. So Typescript didn't like it when I tried to cast one to the other. But I was able to do `statement as unknown as AstNode`, if that's better 🤷‍♂️